homeassistant/components/intellifire/entity.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from datetime import datetime
+
+from intellifire4py import IntellifirePollData
+
+from homeassistant.components.intellifire import IntellifireDataUpdateCoordinator
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from collections.abc import Callable
+from dataclasses import dataclass
+
+
+@dataclass
+class IntellifireEntityDescription(EntityDescription):
+    """mixing with class."""
+
+
+@dataclass
+class IntellifireEntity(CoordinatorEntity):
+    """Define a generic class for Intellifire entities."""
+
+    coordinator: IntellifireDataUpdateCoordinator
+    _attr_attribution = "Data provided by unpublished Intellifire API"
+
+    def __init__(
+            self,
+            coordinator: IntellifireDataUpdateCoordinator,
+            description: IntellifireEntityDescription,
```

Comment:
You can simple use the basic EntityDescription (imported from `homeassistant.helpers.entity`)
```suggestion
            description: EntityDescription,
```